crates/composer/src/main.rs

```diff
@@ -13,25 +16,33 @@ fn main() -> Result<()> {
 
     let (_stream, stream_handle) = OutputStream::try_default()?;
 
+    let jukebox = Jukebox::new().context("creating jukebox")?;
     loop {
-        if let Err(err) = handle_datagram(&socket, &stream_handle) {
+        if let Err(err) = handle_datagram(&socket, &stream_handle, &jukebox) {
             eprintln!("Could not process datagram. Ignoring and continuing. {:?}", err);
         }
     }
 }
 
-fn handle_datagram(socket: &UdpSocket, output_stream: &OutputStreamHandle) -> Result<()> {
+fn handle_datagram(
+    socket: &UdpSocket,
+    output_stream: &OutputStreamHandle,
+    jukebox: &Jukebox,
+) -> Result<()> {
     // Size up to max normal network packet size
     let mut buf = [0; 1500];
     let (number_of_bytes, _) = socket.recv_from(&mut buf)?;
 
     let event: Event = bincode::deserialize(&buf[..number_of_bytes])?;
     println!("Received an event ({number_of_bytes} bytes): {:?}", event);
 
-    // FIXME: do the decoding and file reading outside the loop
-    let file = BufReader::new(File::open("src/sound_samples/click.wav")?);
-    let source = Decoder::new(file)?;
-    output_stream.play_raw(source.convert_samples())?;
+    let sample = match event {
+        Event::TestTick => Sample::Click,
+
+        // TODO(Matej): add different sounds for these, and vary some their quality based on length.
+        Event::StderrWrite { length: _ } | Event::StdoutWrite { length: _ } => Sample::Click,
```

Comment:
@goodhoko FYI the `TODO(author)` notation is a convention we use in tonari codebase. The person mentioned is somewhat counterintuitively _not_ the one who should fix the issue, but the one who added the TODO and who can be reached for more context is necessary.